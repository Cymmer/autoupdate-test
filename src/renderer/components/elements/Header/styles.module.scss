@import '../../../scss/variables';
@import '../../../scss/mixins';
@import '../../../scss/themes';

@mixin children($margin) {
  @include flex-between;
  margin: $margin;

  @include breakpoint(sm) {
    flex-direction: column;
    align-items: flex-start;
  }
}

.Header {
  @include theme {
    border-bottom: 1px solid t('borderColor');
    background-color: t('cardBackgroundColor');
  }
  margin: 0 0 32px 0;
  padding: 24px 0 0 0;

  &_breadcrumbs {
    margin: 0 0 24px 0;
  }

  &_children {
    @include children(0 0 16px 0);

    &___noTabs {
      @include children(0 0 24px 0);
    }
  }
}
