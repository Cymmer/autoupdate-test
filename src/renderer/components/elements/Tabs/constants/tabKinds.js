const tabKinds = {
  BUTTON: 'button',
  LINK: 'link',
};

export default tabKinds;
