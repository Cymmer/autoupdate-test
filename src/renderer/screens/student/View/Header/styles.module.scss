@import '../../../../scss/variables';
@import '../../../../scss/mixins';
@import '../../../../scss/themes';

.Header {
  @include theme {
    border-bottom: 1px solid t('borderColor');
    background-color: t('cardBackgroundColor');
  }
  margin: 0 0 32px 0;

  &_breadcrumbs {
    padding: 24px 0;
  }

  &_children {
    @include flex-between;
    margin: 0 0 16px 0;
  }
}
