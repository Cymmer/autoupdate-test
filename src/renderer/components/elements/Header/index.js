import React from 'react';
import cn from 'classnames';
import PropTypes from 'prop-types';
import styles from './styles.module.scss';

import Tabs from '../Tabs';
import Breadcrumbs from '../Breadcrumbs';
import Container from '../Container';
import tabKinds from '../Tabs/constants/tabKinds';
import tabTypes from '../Tabs/constants/tabTypes';

const Header = ({ id, tabs, activeTab, breadcrumbs, children }) => (
  <div className={styles.Header}>
    <Container>
      {breadcrumbs && (
        <Breadcrumbs
          className={styles.Header_breadcrumbs}
          links={breadcrumbs}
        />
      )}
      <div
        className={cn({
          [styles.Header_children]: tabs !== null,
          [styles.Header_children___noTabs]: tabs === null,
        })}
      >
        {children}
      </div>
      {tabs && (
        <Tabs
          activeTab={activeTab}
          className={styles.Header_tabs}
          data-test="tabs"
          id={id}
          tabs={tabs}
          type={tabTypes.HORIZONTAL.LG}
        />
      )}
    </Container>
  </div>
);

Header.defaultProps = {
  id: null,
  tabs: null,
  activeTab: null,
  breadcrumbs: null,
};

Header.propTypes = {
  id: PropTypes.string,
  tabs: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.string,
      name: PropTypes.string,
      isLocked: PropTypes.bool,
      value: PropTypes.oneOfType([PropTypes.string, PropTypes.number])
        .isRequired,
      kind: PropTypes.oneOf([tabKinds.BUTTON, tabKinds.LINK]),
      // if type is BUTTON -> func, else if type is LINK -> string
      action: PropTypes.oneOfType([PropTypes.func, PropTypes.string]),
    })
  ),
  activeTab: PropTypes.string,
  breadcrumbs: PropTypes.arrayOf(
    PropTypes.shape({
      name: PropTypes.string,
      link: PropTypes.string,
    })
  ),
  children: PropTypes.any.isRequired,
};

export default Header;
