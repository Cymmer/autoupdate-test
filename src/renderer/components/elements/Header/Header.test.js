import React from 'react';
import { shallow } from 'enzyme';
import Header from './index';
import tabKinds from '../Tabs/constants/tabKinds';

describe('RENDERING - Header', () => {
  it('should render with default props', () => {
    const component = shallow(
      <Header>
        <p>Test</p>
      </Header>
    );
    expect(component).toMatchSnapshot();
  });

  it('should render the breadcrumbs if it has breadcrumbs', () => {
    const component = shallow(
      <Header
        breadcrumbs={[
          {
            name: 'Link 1',
            link: 'https://www.codechum.com/link-1',
          },
          {
            name: 'Link 2',
            link: 'https://www.codechum.com/link-2',
          },
        ]}
      >
        <p>Test</p>
      </Header>
    );
    expect(component).toMatchSnapshot();
  });

  it('should render the tabs if it has tabs', () => {
    const component = shallow(
      <Header
        tabs={[
          {
            name: 'Tab 1',
            value: 'tab-1',
            kind: tabKinds.BUTTON,
            action: () => {},
          },
          {
            name: 'Tab 2',
            value: 'tab-2',
            kind: tabKinds.LINK,
            action: 'https://www.codechum.com',
          },
          {
            name: 'Tab 3',
            value: 'tab-3',
            kind: tabKinds.BUTTON,
            action: () => {},
          },
        ]}
      >
        <p>Test</p>
      </Header>
    );
    expect(component).toMatchSnapshot();
  });
});
