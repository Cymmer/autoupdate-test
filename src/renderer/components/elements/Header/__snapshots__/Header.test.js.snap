// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RENDERING - Header should render the breadcrumbs if it has breadcrumbs 1`] = `
<div
  className="Header"
>
  <Container
    className={null}
  >
    <Breadcrumbs
      className="Header_breadcrumbs"
      links={
        Array [
          Object {
            "link": "https://www.codechum.com/link-1",
            "name": "Link 1",
          },
          Object {
            "link": "https://www.codechum.com/link-2",
            "name": "Link 2",
          },
        ]
      }
    />
    <div
      className="Header_children___noTabs"
    >
      <p>
        Test
      </p>
    </div>
  </Container>
</div>
`;

exports[`RENDERING - Header should render the tabs if it has tabs 1`] = `
<div
  className="Header"
>
  <Container
    className={null}
  >
    <div
      className="Header_children"
    >
      <p>
        Test
      </p>
    </div>
    <Tabs
      activeTab={null}
      className="Header_tabs"
      data-test="tabs"
      id={null}
      isLoading={false}
      tabClassName={null}
      tabs={
        Array [
          Object {
            "action": [Function],
            "kind": "button",
            "name": "Tab 1",
            "value": "tab-1",
          },
          Object {
            "action": "https://www.codechum.com",
            "kind": "link",
            "name": "Tab 2",
            "value": "tab-2",
          },
          Object {
            "action": [Function],
            "kind": "button",
            "name": "Tab 3",
            "value": "tab-3",
          },
        ]
      }
      type="horizontal___lg"
    />
  </Container>
</div>
`;

exports[`RENDERING - Header should render with default props 1`] = `
<div
  className="Header"
>
  <Container
    className={null}
  >
    <div
      className="Header_children___noTabs"
    >
      <p>
        Test
      </p>
    </div>
  </Container>
</div>
`;
